src/IECoreArnold/Renderer.cpp

```diff
@@ -873,6 +884,10 @@ class ShaderCache : public IECore::RefCounted
 					// shader.
 					toErase.push_back( it->first );
 				}
+				else
+				{
+					it->second->updateNodeParameters();
```

Comment:
The unit tests prove that this works, but I don't understand why this should not be after the actual node deletion a few lines down. My expectation would be that Arnold nodes haven't been deleted at this point, so `updateNodeParameters()` should still find a camera that is in `toErase`. But I must have something wrong-side up.

Reply:
Not sure I'm following, but there won't ever be any cameras in `toErase`, because we're just iterating over shaders.

Reply:
Right right, I forgot we use a separate cache for shaders. This makes sense to me now.